### PR TITLE
Add RBAC client

### PR DESCRIPTION
**Pull request by epapbak**
# Description

- RBAC Client
- Add authorization middleware in server
- Update UTs (little bit of coupling when it comes to CreateServer method with so much args, one more refactor commit coming)
- Add `rbac` section to configuration files

Part of CCXDEV-12884

# Why this change

After some investigation around [ADR-036](https://docs.google.com/document/d/1LNun1PuvL3EFyfbSWAa6ixUAm-pJoWzQiNJj7hVjkZY/edit?tab=t.0#heading=h.nq1osedn93p8) and a quick discussion with PM, it's been decided that we should support RBAC, mainly because if a consumer of our API uses it with any kind of user, and has configured resource-based access control, we are not enforcing it as of today.

In this change, we introduce the code necessary to handle authorization, but since we don't know which consumers use what and how it can affect them, RBAC is disabled by default and we're doing some extra logging [here](https://github.com/RedHatInsights/insights-results-smart-proxy/pull/1355/files#diff-08316311bca81628634f7ed55949099773edf5af671129d2e6bcbc7dbd1c6825R119) (there's more to come in the next and last PR).

## Type of change

- New feature (non-breaking change which adds functionality)
- Unit tests

## Testing steps

CI

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by Jakub007d — APPROVED** [review]

**Review by Bee-lee — APPROVED** [review]
LGTM overall, ty! would be nice to have some context included, but it is in Jira after all

**Review by JiriPapousek — APPROVED** [review]

File: auth/rbac.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		}
+		return nil
+	}
+	return nil
+}
+
+// requestAccess handles the call(s) to RBAC taking into account that the response
+// is paginated
+func (rc *rbacClientImpl) requestAccess(url, identityToken string) []types.RbacData {
+	//TODO Change return to (rbacData, err) and forward error?
```
Comment (juandspy) [inline auth/rbac.go:89]: leftover?

File: config.toml [hunk 2]
```diff
@@ -67,3 +67,8 @@ database = 0
 endpoint = "localhost:6379"
 password = ""
 timeout_seconds = 30
+
+[rbac]
+host = https://console.redhat.com/api/rbac/v1
```
Comment (juandspy) [inline config.toml:72]: should be string

File: config-devel.toml [hunk 3]
```diff
@@ -52,3 +52,8 @@ database = 0
 endpoint = "localhost:6379"
 password = ""
 timeout_seconds = 30
+
+[rbac]
+host = https://console.redhat.com/api/rbac/v1
```
Comment (juandspy) [inline config-devel.toml:57]: same here

File: auth/rbac.go (hunk 1 above)
Comment (epapbak) [reply auth/rbac.go:89]: probably =). But I have two or three  TODOs in the new code that I'll handle later on

**Pull request merged**